### DISCUSSION TITLE
vz_mini_hacks do not even init (no log appear)

### DISCUSSION THREAD

**Discussion by stepanovdg** [gtxaspec/wz_mini_hacks#639] [Q&A] [answered]
Hi All, 

I am new here. Have firmware version installed demo_wcv3_4.36.9.139.zip using sdcard. Sdcard is usable ( after removing from camera there appear new folders with record and etc.
I tried several times to use wz_mini by coping whatever is SD_ROOT to empty sdcard. 
But nothing changed. Camera is successfully connecting to wifi and qyze servers - i can see video from mobile app. 
But ssh port 22 - connection refused.
I looked through latest discussions and tried to use zip download instead of git clone - still the same.

What can I miss and what to check?

**Comment by gtxaspec** [comment]
post a picture of the root of your sd card in your pc

**Comment by stepanovdg** [comment]
<img width="757" alt="Screenshot 2023-10-17 at 7 36 37 AM" src="https://github.com/gtxaspec/wz_mini_hacks/assets/1208467/75d9a780-c585-4c00-9ccf-3e37a894623d">

**Comment by virmaior** [comment]
1. is there anything in `wz_mini/log/` ?
2. did you set up the ssh key for connecting?
3. is the SD card formatted as fat32 (ExFAT works for recording but doesn't work for wz_mini_hack because the bootloader can't access it).

**Comment by archandanime** [comment]
I think it's very likely to be the third reason @virmaior, he is right that SD card boot image `factory_t31_...` was not even booted because:
- `swap.gz` has not been extracted.
- Can't be SSH public key because if SSH is running, the error would be:
```
Permission denied (publickey)
```
- `wz_mini/log/` is empty because it has same modification timestamp with other directories.

**Comment by stepanovdg** [comment]
Nothing in the log. 
Card formatted to FAT32 used that guide https://tinymachines3d.freshdesk.com/support/solutions/articles/33000259849-fat-32-4kb-formatting-for-mac-users 
retried again just now

sudo newfs_msdos -F 32 -c 8 -v TM3D /dev/disk5
Password:
warning: /dev/disk5 is not a character device
512 bytes per physical sector
/dev/disk5: 15571216 sectors in 1946402 FAT32 clusters (4096 bytes/cluster)
bps=512 spc=8 res=32 nft=2 mid=0xf0 spt=32 hds=255 hid=0 drv=0x00 bsec=15601664 bspf=15207 rdcl=2 infs=1 bkbs=6

still camera loaded - connected to cloud but no ssh -> 

$ ssh root@10.5.5.52
ssh: connect to host 10.5.5.52 port 22: Connection refused
$ nmap -n 10.5.5.52/32
Starting Nmap 7.93 ( https://nmap.org ) at 2023-10-18 16:02 PDT
Nmap scan report for 10.5.5.52
Host is up (0.0065s latency).
All 1000 scanned ports on 10.5.5.52 are in ignored states.
Not shown: 1000 closed tcp ports (conn-refused)

Nmap done: 1 IP address (1 host up) scanned in 1.87 seconds

**Comment by stepanovdg** [comment]
public key line is added.

**Comment by archandanime** [comment]
That is really weird. The only reason I could think of is that Wyze removed the ability to boot kernel image from SD card in your(new?) unit like the Pan Cam v3. I could be very wrong but I can't be sure without seeing the boot log over serial connection. If the uboot ever sees the boot image the log would look like:
```
reading factory_t31_ZMC6tiIDQN
reading factory_t31_ZMC6tiIDQN
## Booting kernel from Legacy Image at 84000000 ...
   Image Name:   Linux-3.10.14__isvp_swan_1.0__
   Image Type:   MIPS Linux Kernel Image (lzma compressed)
   Data Size:    2387173 Bytes = 2.3 MiB
   Load Address: 80010000
   Entry Point:  803d3c50
   Verifying Checksum ... OK
   Uncompressing Kernel Image ... OK
   ```
If the SD card boot image was ever booted, `wz_mini_initramfs.log` should be there but there is no initramfs log either so you are right(still not 100% sure) that the boot image `factory_t31_ZMC6tiIDQN` wasn't even booted at all.

**Comment by stepanovdg** [comment]
I have wyze cam v3 not pan - it was pretty old ( was on shelf for 1 year) - so when i just power  - it was offline and wasnt able to connect to cloud by using setup - always connection timeout. 
So I first download latest firmware 4.36.10.4054 and flash it through the same sdcard with factory reset. After I was able to connect camera to cloud. 
So I start looking how to connect camera to octopi mainsail and realise I need rtsp - thats how I found this repo. 
I read manual and flashed camera with 4.36.9.139 - this version can be see in cloud ( through mobile application)
But after putting mini hacs on sd card they never loaded. 

So do you have any manual how to get serial connection ? 
Just plug in to linux usb and connect through screen to device with some badrate? 
Or i need to open camera and solder usb programmer to some pinouts?

**Reply by gtxaspec** [reply]
open the camera and hookup a serial interface to these pins:
left to right:
GND/TX/RX
do not hookup 5v or 3v from the serial interface, just the pins above

![image](https://github.com/gtxaspec/wz_mini_hacks/assets/12115272/97069654-deb2-4936-87ad-ac9588946e62)

**Reply by archandanime** [reply]
On the photo:
Camera:..... GND  TXD  RXD
...................... |...... |......|
USB UART   GND   RXD TXD
Command to run(if you are on Linux)
```
sudo picocom /dev/ttyUSB0 -b 115200 -l | tee wcv3_boot.log
```
if there is still no output, try swapping TXD and RXD wires.
And don't forget to remove your SD card first or the SD card would be broken or worse the SD card slot is damaged.

**Comment by stepanovdg** [comment]
[boot_without_card.log](https://github.com/gtxaspec/wz_mini_hacks/files/13048347/boot_without_card.log)
[wcv3_boot.log](https://github.com/gtxaspec/wz_mini_hacks/files/13048348/wcv3_boot.log)
(( Last week broke sdcard in raspberry why putting in case. 
Today was sure will not be idiot again and now sdcard will be safe.
Spent too many time strugling with rubber caps and forgot to remove sdcard - broke (

Also do not found male pins so soldered wires to programmer and camera. 
Attached 2 logs with or without card.

**Reply by gtxaspec** [reply]
looks like memory card is not formatted correctly:
```
** Partition 1 not valid on device 0 **
fat_register_device fail
```
or is corrupted

**Reply by stepanovdg** [reply]
used new card 32gb formatted by 
sudo newfs_msdos -F 32 -c 8 -v TM3D /dev/disk4
then mount and copy files
after unmount and put into camera.

**Reply by stepanovdg** [reply]
Volume name : TM3D
Volume type : Unknown
BSD device node : disk4
Mount point : /Volumes/TM3D
File system : MS-DOS (FAT32)
Connection : Secure Digital
Device tree path : IODeviceTree:/arm-io/apcie@80000000/pci-bridge1@1/pcie-sdreader@0
Writable : Yes
Is case-sensitive : No
File system UUID : 74C68938-A09F-3765-A300-9D6F93298DDB
Volume capacity : 31,044,308,992
Available space (Purgeable + Free) : 30,948,405,248
Purgeable space : 0
Free space : 30,948,405,248
Used space : 95,903,744
Owners enabled : No
Is encrypted : No
System Integrity Protection supported : No
Can be verified : Yes
Can be repaired : Yes
Bootable : No
Journaled : No
Disk number : 4
Media name : Apple SDXC Reader Media
Media type : Generic
Ejectable : Yes
Solid state : No
S.M.A.R.T. status : Not Supported

**Reply by gtxaspec** [reply]
looks like the partition is msdos, but the partition table is GPT judging by the UUID... you need MBR

**Reply by stepanovdg** [reply, marked as answer]
sudo diskutil eraseDisk FAT32 wyze-mini MBRFormat /dev/disk4
Password:
Started erase on disk4 (TM3D)
Unmounting disk
Creating the partition map
Waiting for partitions to activate
Formatting disk4s1 as MS-DOS (FAT32) with name wyze-mini
512 bytes per physical sector
/dev/rdisk4s1: 60713984 sectors in 1897312 FAT32 clusters (16384 bytes/cluster)
bps=512 spc=32 res=32 nft=2 mid=0xf8 spt=32 hds=255 hid=8192 drv=0x80 bsec=60743680 bspf=14823 rdcl=2 infs=1 bkbs=6
Mounting disk
Finished erase on disk4


and now Volume name : WYZE-MINI
Volume type : Unknown
BSD device node : disk4s1
Mount point : /Volumes/WYZE-MINI
File system : MS-DOS (FAT32)
Connection : Secure Digital
Device tree path : IODeviceTree:/arm-io/apcie@80000000/pci-bridge1@1/pcie-sdreader@0
Writable : Yes
Is case-sensitive : No
File system UUID : 6DC75BA6-580A-3160-8A88-755E0768C328
Volume capacity : 31,085,576,192
Available space (Purgeable + Free) : 31,083,839,488
Purgeable space : 0
Free space : 31,083,839,488
Used space : 1,736,704
Owners enabled : No
Is encrypted : No
System Integrity Protection supported : No
Can be verified : Yes
Can be repaired : Yes
Bootable : No
Journaled : No
Disk number : 4
Partition number : 1
Media name : 
Media type : Generic
Ejectable : Yes
Solid state : No
S.M.A.R.T. status : Not Supported
Parent disks : disk4


Thats worked
Just after was require to setup again as it was dropping wifi due to reason 3


Thank you very much for your help!
